models/DataObject/ClassDefinition/Data.php

```diff
@@ -1232,12 +1232,18 @@ public function adoptMainDefinition(DataObject\ClassDefinition\Data $mainDefinit
         }
     }
 
-    public function appendData(?array $existingData, array $additionalData): ?array
+    /**
+     * @throws \LogicException
+     */
+    public function appendData(?array $existingData, array $additionalData): array
     {
         return $existingData;
     }
 
-    public function removeData(mixed $existingData, mixed $removeData): mixed
+    /**
+     * @throws \LogicException
+     */
+    public function removeData(?array $existingData, array $removeData): array
```

Comment:
looks fine for return type, but for parameters, should we keep `mixed` and trigger a deprecation first?

Reply:
@kingjia90 mixed isn't possible, because of the usage of foreach, right? Or did you mean when someone overwrite the method? Than this should also possible with mixed instead of array, right?

Reply:
Yes, i was thinking about overwritten methods, devs won't be able to detect this change and correct it accordingly, but doesn't seem a big deal.

Reply:
But when the dev have a overwrite method, a deprecation warning in this method won't work either.

Reply:
Mhh true, then nevermind, it should be fine as is